### PR TITLE
HBASE-24148: Upgrade Thrift to 0.13.0

### PR DESCRIPTION
**Pull request by tamaashu**
Upgrade thrift, adopt Java code where needed.

Change-Id: I8d78f6bd04ed3c7787b8f9beaf370dfdac9574cb

### CODE REVIEW

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 30s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 44s |  Maven dependency ordering for branch  |
| -1 :x: |  mvninstall  |   4m 16s |  root in master failed.  |
| -0 :warning: |  checkstyle  |   2m  2s |  The patch fails to run checkstyle in root  |
| -1 :x: |  spotbugs  |   7m  4s |  root in master failed.  |
| -1 :x: |  spotbugs  |   0m 21s |  hbase-mapreduce in master failed.  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 19s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   5m  1s |  the patch passed  |
| -0 :warning: |  checkstyle  |   2m 19s |  root: The patch generated 1 new + 0 unchanged - 0 fixed = 1 total (was 0)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  xml  |   0m  1s |  The patch has no ill-formed XML file.  |
| +1 :green_heart: |  hadoopcheck  |  10m 22s |  Patch does not cause any errors with Hadoop 2.10.0 or 3.1.2.  |
| +1 :green_heart: |  spotbugs  |  13m  4s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 42s |  The patch does not generate ASF License warnings.  |
|  |   |  55m  8s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.8 Server=19.03.8 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1462/1/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/1462 |
| JIRA Issue | HBASE-24148 |
| Optional Tests | dupname asflicense spotbugs hadoopcheck hbaseanti checkstyle xml |
| uname | Linux 51ee3e438b13 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / eb7df0498c |
| mvninstall | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1462/1/artifact/yetus-general-check/output/branch-mvninstall-root.txt |
| checkstyle | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1462/1/artifact/yetus-general-check/output//home/jenkins/jenkins-slave/workspace/Base-PreCommit-GitHub-PR_PR-1462@2/yetus-general-check/output/maven-branch-checkstyle-root.txt |
| spotbugs | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1462/1/artifact/yetus-general-check/output/branch-spotbugs-root.txt |
| spotbugs | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1462/1/artifact/yetus-general-check/output/branch-spotbugs-hbase-mapreduce.txt |
| checkstyle | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1462/1/artifact/yetus-general-check/output/diff-checkstyle-root.txt |
| Max. process+thread count | 93 (vs. ulimit of 10000) |
| modules | C: hbase-mapreduce hbase-thrift . U: . |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1462/1/console |
| versions | git=2.17.1 maven=(cecedd343002696d0abb50b32b541b8a6ba2883f) spotbugs=3.1.12 |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 33s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  2s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 43s |  Maven dependency ordering for branch  |
| -1 :x: |  mvninstall  |   4m 18s |  root in master failed.  |
| -1 :x: |  compile  |   1m 48s |  root in master failed.  |
| -1 :x: |  shadedjars  |   3m 55s |  branch has 16 errors when building our shaded downstream artifacts.  |
| -0 :warning: |  javadoc  |   1m 57s |  root in master failed.  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 20s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   5m  6s |  the patch passed  |
| +1 :green_heart: |  compile  |   2m 51s |  the patch passed  |
| +1 :green_heart: |  javac  |   2m 51s |  root generated 0 new + 34 unchanged - 4 fixed = 34 total (was 38)  |
| +1 :green_heart: |  shadedjars  |   4m 28s |  patch has no errors when building our shaded downstream artifacts.  |
| -0 :warning: |  javadoc  |   2m 36s |  root generated 72 new + 15 unchanged - 0 fixed = 87 total (was 15)  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  | 163m 21s |  root in the patch passed.  |
|  |   | 196m 18s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.8 Server=19.03.8 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1462/1/artifact/yetus-jdk8-hadoop2-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/1462 |
| JIRA Issue | HBASE-24148 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux a7e449387982 4.15.0-60-generic #67-Ubuntu SMP Thu Aug 22 16:55:30 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / eb7df0498c |
| Default Java | 1.8.0_232 |
| mvninstall | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1462/1/artifact/yetus-jdk8-hadoop2-check/output/branch-mvninstall-root.txt |
| compile | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1462/1/artifact/yetus-jdk8-hadoop2-check/output/branch-compile-root.txt |
| shadedjars | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1462/1/artifact/yetus-jdk8-hadoop2-check/output/branch-shadedjars.txt |
| javadoc | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1462/1/artifact/yetus-jdk8-hadoop2-check/output/branch-javadoc-root.txt |
| javadoc | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1462/1/artifact/yetus-jdk8-hadoop2-check/output/diff-javadoc-javadoc-root.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1462/1/testReport/ |
| Max. process+thread count | 4446 (vs. ulimit of 10000) |
| modules | C: hbase-mapreduce hbase-thrift . U: . |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1462/1/console |
| versions | git=2.17.1 maven=(cecedd343002696d0abb50b32b541b8a6ba2883f) |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m  9s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 39s |  Maven dependency ordering for branch  |
| -1 :x: |  mvninstall  |   5m 42s |  root in master failed.  |
| -1 :x: |  compile  |   2m 36s |  root in master failed.  |
| -1 :x: |  shadedjars  |   4m 59s |  branch has 16 errors when building our shaded downstream artifacts.  |
| -0 :warning: |  javadoc  |   0m 19s |  root in master failed.  |
| -0 :warning: |  javadoc  |   0m 18s |  hbase-mapreduce in master failed.  |
| -0 :warning: |  javadoc  |   0m 56s |  hbase-thrift in master failed.  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 18s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   6m 47s |  the patch passed  |
| +1 :green_heart: |  compile  |   4m  0s |  the patch passed  |
| +1 :green_heart: |  javac  |   4m  0s |  root generated 0 new + 34 unchanged - 4 fixed = 34 total (was 38)  |
| +1 :green_heart: |  shadedjars  |   5m 42s |  patch has no errors when building our shaded downstream artifacts.  |
| -0 :warning: |  javadoc  |   0m 19s |  hbase-mapreduce in the patch failed.  |
| -0 :warning: |  javadoc  |   0m 56s |  hbase-thrift in the patch failed.  |
| -0 :warning: |  javadoc  |   0m 18s |  root in the patch failed.  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  | 247m 43s |  root in the patch passed.  |
|  |   | 285m 10s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.8 Server=19.03.8 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1462/1/artifact/yetus-jdk11-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/1462 |
| JIRA Issue | HBASE-24148 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux ebb093013a07 4.15.0-91-generic #92-Ubuntu SMP Fri Feb 28 11:09:48 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / eb7df0498c |
| Default Java | 2020-01-14 |
| mvninstall | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1462/1/artifact/yetus-jdk11-hadoop3-check/output/branch-mvninstall-root.txt |
| compile | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1462/1/artifact/yetus-jdk11-hadoop3-check/output/branch-compile-root.txt |
| shadedjars | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1462/1/artifact/yetus-jdk11-hadoop3-check/output/branch-shadedjars.txt |
| javadoc | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1462/1/artifact/yetus-jdk11-hadoop3-check/output/branch-javadoc-root.txt |
| javadoc | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1462/1/artifact/yetus-jdk11-hadoop3-check/output/branch-javadoc-hbase-mapreduce.txt |
| javadoc | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1462/1/artifact/yetus-jdk11-hadoop3-check/output/branch-javadoc-hbase-thrift.txt |
| javadoc | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1462/1/artifact/yetus-jdk11-hadoop3-check/output/patch-javadoc-hbase-mapreduce.txt |
| javadoc | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1462/1/artifact/yetus-jdk11-hadoop3-check/output/patch-javadoc-hbase-thrift.txt |
| javadoc | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1462/1/artifact/yetus-jdk11-hadoop3-check/output/patch-javadoc-root.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1462/1/testReport/ |
| Max. process+thread count | 3383 (vs. ulimit of 10000) |
| modules | C: hbase-mapreduce hbase-thrift . U: . |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1462/1/console |
| versions | git=2.17.1 maven=(cecedd343002696d0abb50b32b541b8a6ba2883f) |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

File: hbase-mapreduce/src/test/java/org/apache/hadoop/hbase/snapshot/TestExportSnapshotAdjunct.java [hunk 1]
```diff
@@ -119,8 +119,7 @@ public void testExportRetry() throws Exception {
     conf.setBoolean(ExportSnapshot.Testing.CONF_TEST_FAILURE, true);
     conf.setInt(ExportSnapshot.Testing.CONF_TEST_FAILURE_COUNT, 2);
     conf.setInt("mapreduce.map.maxattempts", 3);
-    TestExportSnapshot.testExportFileSystemState(conf, tableName,
-      Bytes.toBytes(snapshotName), Bytes.toBytes(snapshotName),
+    TestExportSnapshot.testExportFileSystemState(conf, tableName, snapshotName, snapshotName,
```
Comment (virajjasani) [inline hbase-mapreduce/src/test/java/org/apache/hadoop/hbase/snapshot/TestExportSnapshotAdjunct.java:122]: Please resolve conflicts, this one is taken care of.
Thanks

**Comment by saintstack** [comment]
Closing. Wrapped @tamaashu 's patch into a fatter one in a new PR that includes regeneration of java files from thrift IDLs.

**Pull request closed without merging**